compiler/optimizer/VPHandlers.cpp

```diff
@@ -2777,6 +2777,7 @@ TR::Node *constrainIaload(OMR::ValuePropagation *vp, TR::Node *node)
    if (base && base->getClass() && !vp->comp()->getOption(TR_DisableMarkingOfHotFields) &&
       node->getFirstChild()->getOpCode().hasSymbolReference() &&
       node->getFirstChild()->getSymbol()->isCollectedReference() &&
+      !vp->comp()->fej9()->isHotReferenceFieldRequired() &&
```

Comment:
This doesn't make a lot of sense - why are we going to mark hot fields if they are not required? is the GC even going to consume the data generated using this old API?

Reply:
This line is needed so that hot fields are not marked via this old hot field marking mechanism when dynamic breadth first scan ordering is enabled. On my to-do list over the next month will be to remove all the old hot field remnants along with the hot field alignment code as we have briefly discussed in the past. This line will be removed when the old hot field code is removed approximately in the next month.